src/main/java/org/killbill/billing/plugin/dwolla/api/DwollaPaymentPluginApi.java

```diff
@@ -107,6 +114,57 @@ protected String getPaymentMethodId(DwollaPaymentMethodsRecord input) {
     }
 
     @Override
+    public void addPaymentMethod(UUID kbAccountId, UUID kbPaymentMethodId, PaymentMethodPlugin paymentMethodProps, boolean setDefault, Iterable<PluginProperty> properties, CallContext context) throws PaymentPluginApiException {
+        Map<String, String> mergedProperties = checkForDwollaDirectSolution(paymentMethodProps, properties, context);
+        final DateTime utcNow = clock.getUTCNow();
+        try {
+            dao.addPaymentMethod(kbAccountId, kbPaymentMethodId, setDefault, mergedProperties, utcNow, context.getTenantId());
+        } catch (final SQLException e) {
+            throw new PaymentPluginApiException("Unable to add payment method for kbPaymentMethodId " + kbPaymentMethodId, e);
```

Comment:
Nit: better to use a key/value format for tools like Splunk (to automatically extract fields), e.g.

```
throw new PaymentPluginApiException("Unable to add payment method for kbPaymentMethodId='" + kbPaymentMethodId + "'", e);
```